### PR TITLE
Refactoring & Test Cleanup

### PR DESCRIPTION
**Pull request by k-xo**
- Updates test block number
- Clean up tests
- Removes the unused PSM Router

### CODE REVIEW

**Comment by ElliotFriedman** [comment]
Once you're ready for a review on this PR, please ping

File: contracts/test/integration/IntegrationTestPCVGuardian.t.sol [hunk 1]
```diff
@@ -26,7 +26,7 @@ contract IntegrationTestPCVGuardian is DSTest {
     IVolt private fei = IVolt(0x956F47F50A910163D8BF957Cf5846D573E7f87CA);
 
     IPCVDepositTest private pcvDeposit =
-        IPCVDepositTest(0x4188fbD7aDC72853E3275F1c3503E170994888D7);
+        IPCVDepositTest(0x985f9C331a9E4447C782B98D6693F5c7dF8e560e);
```
Comment (ElliotFriedman) [inline contracts/test/integration/IntegrationTestPCVGuardian.t.sol:29]: This address should be pulled in from a solidity config file like Fixtures.sol

File: contracts/test/integration/IntegrationTestVoltDeployment.t.sol [hunk 2]
```diff
@@ -198,9 +201,11 @@ contract IntegrationTestVoltDeployment is DSTest, StdLib {
...
 
         assertEq(endingUserVoltBalance - startingUserVoltBalance, mintAmount);
-        assertEq(
-            endingPCVDepositFeiBalance - startingPCVDepositFeiBalance,
-            mintAmount - 1 /// goes down by 1 because of cToken pricing rounding down
+        assertApproxEq(
+            (endingPCVDepositFeiBalance - startingPCVDepositFeiBalance)
+                .toInt256(),
+            (mintAmount - 1).toInt256(), /// goes down by 1 because of cToken pricing rounding down
+            0
```
Comment (ElliotFriedman) [inline contracts/test/integration/IntegrationTestVoltDeployment.t.sol:208]: why do we need to assertApproxEq if we just use 0 basis points? it feels like we can just revert this change.

File: contracts/test/integration/IntegrationTestVoltDeployment.t.sol [hunk 3]
```diff
@@ -14,13 +14,16 @@ import {ERC20CompoundPCVDeposit} from "../../pcv/compound/ERC20CompoundPCVDeposi
...
+import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+
 // Create Core
 // Global Rate Limited Minter
 // Oracle System
 // - Scaling Price Oracle
 // - Oracle Pass Through
 
 contract IntegrationTestVoltDeployment is DSTest, StdLib {
+    using SafeCast for *;
```
Comment (ElliotFriedman) [inline contracts/test/integration/IntegrationTestVoltDeployment.t.sol:26]: let's revert this change if possible

File: contracts/test/integration/IntegrationTestVoltDeployment.t.sol [hunk 4]
```diff
@@ -14,13 +14,16 @@ import {ERC20CompoundPCVDeposit} from "../../pcv/compound/ERC20CompoundPCVDeposi
 import {getCore, getAddresses, FeiTestAddresses} from "./../unit/utils/Fixtures.sol";
 import {NonCustodialPSM, GlobalRateLimitedMinter} from "./../../peg/NonCustodialPSM.sol";
 
+import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
```
Comment (ElliotFriedman) [inline contracts/test/integration/IntegrationTestVoltDeployment.t.sol:17]: let's revert this change if possible

File: contracts/test/integration/IntegrationTestPriceBoundPSMUSDC.t.sol [hunk 5]
```diff
@@ -136,12 +148,22 @@ contract IntegrationTestPriceBoundPSMUSDCTest is DSTest {
...
+        uint256 amountUSDCIn = 100e18;
+
+        uint256 currentPegPrice = oracle.getCurrentOraclePrice();
+
+        uint256 fee = (amountUSDCIn.mul(1e12) * psm.mintFeeBasisPoints()) /
+            Constants.BASIS_POINTS_GRANULARITY;
+
+        uint256 amountOut = ((amountUSDCIn.mul(1e18) / currentPegPrice)).mul(
+            1e12
+        ) - fee;
```
Comment (ElliotFriedman) [inline contracts/test/integration/IntegrationTestPriceBoundPSMUSDC.t.sol:160]: there should be a comment here explaining all the things that are happening and why

File: contracts/test/integration/IntegrationTestPriceBoundPSMUSDC.t.sol [hunk 6]
```diff
@@ -115,8 +116,19 @@ contract IntegrationTestPriceBoundPSMUSDCTest is DSTest {
...
     function testGetRedeemAmountOut() public {
-        uint256 amountVoltIn = 100e12;
-        assertEq(psm.getRedeemAmountOut(amountVoltIn), 101);
+        uint256 amountVoltIn = 100e18;
+
+        uint256 currentPegPrice = oracle.getCurrentOraclePrice().div(1e12);
+        uint256 fee = (amountVoltIn * psm.redeemFeeBasisPoints()) /
+            Constants.BASIS_POINTS_GRANULARITY;
+
+        uint256 amountOut = ((amountVoltIn * currentPegPrice).div(1e18)) - fee;
```
Comment (ElliotFriedman) [inline contracts/test/integration/IntegrationTestPriceBoundPSMUSDC.t.sol:125]: there should be a comment here explaining all the things that are happening and why

File: contracts/test/integration/IntegrationTestPriceBoundPSMUSDC.t.sol [hunk 7]
```diff
@@ -115,8 +116,19 @@ contract IntegrationTestPriceBoundPSMUSDCTest is DSTest {
...
+
+        assertApproxEq(
+            psm.getRedeemAmountOut(amountVoltIn).toInt256(),
+            amountOut.toInt256(),
+            1
```
Comment (ElliotFriedman) [inline contracts/test/integration/IntegrationTestPriceBoundPSMUSDC.t.sol:130]: theoretically the math we did here should match up exactly with the numbers the PSM arrives at, so basis point deviation should be 0, so we should be able to use assertEq

File: contracts/test/integration/IntegrationTestPriceBoundPSM.t.sol [hunk 8]
```diff
@@ -138,20 +148,22 @@ contract IntegrationTestPriceBoundPSMTest is DSTest {
...
+        uint256 fee = (mintAmount * psm.mintFeeBasisPoints()) /
+            Constants.BASIS_POINTS_GRANULARITY;
+
+        uint256 amountOut = ((mintAmount.mul(1e18) / currentPegPrice)) - fee;
 
-        assertEq(psm.getRedeemAmountOut(amountVoltIn), expectedAmountStableOut);
+        assertApproxEq(
+            psm.getMintAmountOut(mintAmount).toInt256(),
+            amountOut.toInt256(),
+            1
```
Comment (ElliotFriedman) [inline contracts/test/integration/IntegrationTestPriceBoundPSM.t.sol:161]: theoretically the math we did here should match up exactly with the numbers the PSM arrives at, so basis point deviation should be 0, so we should be able to use assertEq

**Review by thomas-waite — COMMENTED** [review]
Like the idea of removing these unused contracts and cleaning things up. Some comments on the testing changes

File: contracts/test/integration/IntegrationTestVoltDeployment.t.sol (hunk 2 above)
Comment (k-xo) [reply contracts/test/integration/IntegrationTestVoltDeployment.t.sol:208]: noted, fixed

File: contracts/test/integration/IntegrationTestPriceBoundPSM.t.sol [hunk 9]
```diff
@@ -117,8 +118,17 @@ contract IntegrationTestPriceBoundPSMTest is DSTest {
...
     function testGetRedeemAmountOut() public {
-        uint256 amountFeiIn = 100;
-        assertEq(psm.getRedeemAmountOut(amountFeiIn), 101);
+        uint256 currentPegPrice = oracle.getCurrentOraclePrice();
+        uint256 fee = (mintAmount * psm.redeemFeeBasisPoints()) /
+            Constants.BASIS_POINTS_GRANULARITY;
+
+        uint256 amountOut = ((mintAmount * currentPegPrice).div(1e18)) - fee;
+
+        assertApproxEq(
```
[the remark this answers is not in the document]
Comment (k-xo) [reply contracts/test/integration/IntegrationTestPriceBoundPSM.t.sol:125]: The math flow used, is a bit different from the one used in the contract, so there's some slight deviation when calculating this part, which stems from rounding differences, so we just make sure its a maximum of a 1 basis point deviation

File: contracts/test/integration/IntegrationTestPriceBoundPSM.t.sol [hunk 10]
```diff
@@ -19,8 +19,12 @@ import {ERC20CompoundPCVDeposit} from "../../pcv/compound/ERC20CompoundPCVDeposi
 import {Vm} from "./../unit/utils/Vm.sol";
 import {DSTest} from "./../unit/utils/DSTest.sol";
 
+import {Constants} from "../../Constants.sol";
+import "@openzeppelin/contracts/utils/math/SafeMath.sol";
+
 contract IntegrationTestPriceBoundPSMTest is DSTest {
     using SafeCast for *;
+    using SafeMath for uint256;
```
[the remark this answers is not in the document]
Comment (k-xo) [reply contracts/test/integration/IntegrationTestPriceBoundPSM.t.sol]: true! i think using 'mul' & 'div' function swas just nice at the time, but changed this ^_^

File: contracts/test/integration/IntegrationTestPriceBoundPSMUSDC.t.sol [hunk 11]
```diff
@@ -136,12 +148,22 @@ contract IntegrationTestPriceBoundPSMUSDCTest is DSTest {
...
+
+        uint256 fee = (amountUSDCIn.mul(1e12) * psm.mintFeeBasisPoints()) /
+            Constants.BASIS_POINTS_GRANULARITY;
+
+        uint256 amountOut = ((amountUSDCIn.mul(1e18) / currentPegPrice)).mul(
+            1e12
+        ) - fee;
+
         assertApproxEq(
             psm.getMintAmountOut(amountUSDCIn).toInt256(),
```
[the remark this answers is not in the document]
Comment (k-xo) [reply contracts/test/integration/IntegrationTestPriceBoundPSMUSDC.t.sol:163]: `assertAproxEq` only accepts Int256, hence why we do the conversion

File: contracts/test/unit/utils/Fixtures.sol [hunk 12]
```diff
@@ -57,7 +57,7 @@ function getMainnetAddresses() pure returns (FeiTestAddresses memory) {
         ),
         minterAddress: address(0xd51dbA7a94e1adEa403553A8235C302cEbF41a3c),
         burnerAddress: address(0x10),
-        guardianAddress: address(0xEC7AD284f7Ad256b64c6E69b84Eb0F48f42e8196),
+        guardianAddress: address(0x2c2b362e6ae0F080F39b90Cb5657E5550090D6C3),
```
[the remark this answers is not in the document]
Comment (k-xo) [reply contracts/test/unit/utils/Fixtures.sol:60]: No, the address used actually isn't a guardian, so updated this to use the `PCVguardian` address as the guardian address

File: contracts/test/integration/IntegrationTestPCVGuardian.t.sol [hunk 13]
```diff
@@ -26,7 +26,7 @@ contract IntegrationTestPCVGuardian is DSTest {
     IVolt private fei = IVolt(0x956F47F50A910163D8BF957Cf5846D573E7f87CA);
```
[the remark this answers is not in the document]
Comment (k-xo) [reply contracts/test/integration/IntegrationTestPCVGuardian.t.sol:26]: I think I really like this approach, adding more addresses to the current test addresses setup will run into a stack to deep error, and will need some work around for that at some point, so I think this flow is cleaner & also solves for that

[the remark this answers is not in the document]
Comment (ElliotFriedman) [reply contracts/test/integration/IntegrationTestPCVGuardian.t.sol:26]: Let's try out this approach and see if we can get around the stack too deep error by using a library constant instead of the current setup. https://github.com/fei-protocol/fei-protocol-core/blob/develop/contracts/test/integration/fixtures/MainnetAddresses.sol

File: contracts/test/integration/IntegrationTestPriceBoundPSM.t.sol [hunk 14]
```diff
@@ -117,16 +115,25 @@ contract IntegrationTestPriceBoundPSMTest is DSTest {
...
+        uint256 currentPegPrice = oracle.getCurrentOraclePrice();
+        uint256 fee = (mintAmount * psm.redeemFeeBasisPoints()) /
+            Constants.BASIS_POINTS_GRANULARITY;
+
+        uint256 amountOut = ((mintAmount * currentPegPrice) / 1e18) - fee;
+
+        assertApproxEq(
+            psm.getRedeemAmountOut(mintAmount).toInt256(),
+            amountOut.toInt256(),
+            1
```
Comment (ElliotFriedman) [inline contracts/test/integration/IntegrationTestPriceBoundPSM.t.sol:127]: we should be able to change to 0 basis points

File: contracts/test/integration/IntegrationTestPriceBoundPSM.t.sol [hunk 15]
```diff
@@ -138,20 +145,22 @@ contract IntegrationTestPriceBoundPSMTest is DSTest {
...
+        uint256 fee = (mintAmount * psm.mintFeeBasisPoints()) /
+            Constants.BASIS_POINTS_GRANULARITY;
+
+        uint256 amountOut = (((mintAmount * 1e18) / currentPegPrice)) - fee;
 
-        assertEq(psm.getRedeemAmountOut(amountVoltIn), expectedAmountStableOut);
+        assertApproxEq(
+            psm.getMintAmountOut(mintAmount).toInt256(),
+            amountOut.toInt256(),
+            1
```
Comment (ElliotFriedman) [inline contracts/test/integration/IntegrationTestPriceBoundPSM.t.sol:158]: does 0 basis points work here?

File: contracts/test/integration/IntegrationTestPriceBoundPSMUSDC.t.sol [hunk 16]
```diff
@@ -136,12 +144,24 @@ contract IntegrationTestPriceBoundPSMUSDCTest is DSTest {
...
+
+        uint256 amountOut = (((amountUSDCIn * 1e18) / currentPegPrice)) *
+            1e12 -
+            fee;
+
         assertApproxEq(
             psm.getMintAmountOut(amountUSDCIn).toInt256(),
-            9809e10,
+            amountOut.toInt256(),
             1
```
Comment (ElliotFriedman) [inline contracts/test/integration/IntegrationTestPriceBoundPSMUSDC.t.sol:163]: 0 basis points?

File: contracts/test/integration/fixtures/MainnetAddresses.sol [hunk 17]
```diff
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+library MainnetAddresses {
+    // ---------- INTERNAL ADDRESSES ----------
```
Comment (ElliotFriedman) [inline contracts/test/integration/fixtures/MainnetAddresses.sol:5]: Would change the name to `VOLT ADDRESSES`

**Review by ElliotFriedman — APPROVED** [review]
